api/client/image/build.go

```diff
@@ -211,6 +227,9 @@ func runBuild(dockerCli *client.DockerCli, options buildOptions) error {
 
 	// Setup an upload progress bar
 	progressOutput := streamformatter.NewStreamFormatter().NewProgressOutput(progBuff, true)
+	if !dockerCli.IsTerminalOut() {
```

Comment:
My bad then. I think `&lastProgressOutput{progressOutput}` is better. Sorry for the back and forth.